keycloak/openid_client.go

```diff
@@ -25,12 +25,25 @@ type OpenidClient struct {
 	PublicClient bool `json:"publicClient"`
 	BearerOnly   bool `json:"bearerOnly"`
 
+	StandardFlowEnabled       bool `json:"standardFlowEnabled"`
+	ImplicitFlowEnabled       bool `json:"implicitFlowEnabled"`
+	DirectAccessGrantsEnabled bool `json:"directAccessGrantsEnabled"`
+	ServiceAccountsEnabled    bool `json:"serviceAccountsEnabled"`
+
 	ValidRedirectUris []string `json:"redirectUris"`
 }
 
 func (keycloakClient *KeycloakClient) ValidateOpenidClient(client *OpenidClient) error {
-	if !client.BearerOnly && len(client.ValidRedirectUris) == 0 {
-		return fmt.Errorf("validation error: must specify at least one valid redirect uri if access type is PUBLIC or CONFIDENTIAL")
+	if client.BearerOnly && (client.StandardFlowEnabled || client.ImplicitFlowEnabled || client.DirectAccessGrantsEnabled || client.ServiceAccountsEnabled) {
+		return fmt.Errorf("validation error: bearer-only clients cannot issue tokens; no oauth2 flows can be enabled for this client")
```

Comment:
>bearer-only clients cannot issue tokens

It's Keycloak that issues the tokens to clients, not so much clients making tokens themselves.

Reply:
yeah the wording is a bit off here, it should probably be something like

> Keycloak cannot issue tokens for bearer-only clients